### PR TITLE
Adding few modules and components. Styling pending

### PR DESCRIPTION
**Pull request by rpendlimarri**
New changes:
- Home component added
- Filter bar component added
- Table with sample/ temporary data added. Sorting, pagination functionalities added to the table.
- Compare button with a modal dialog pop-up component added
- Data transfer between the component and the modal dialog established
- Required interfaces and models and created
- Fixed the comments received as part of the PR review from Maksym Kondratenko

### CODE REVIEW

File: src/app/app.component.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.main-page-layout {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    padding: 0px;
+    gap: 10px;
+    background-color: rgb(146, 134, 134);
+
```
Comment (miramax-88) [inline src/app/app.component.scss]: please remove empty line

File: src/app/home/filter-bar/filter-bar.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<!-- <p>filter-bar works!</p> -->
+<div class="filter-bar-section">
+    <!-- <p>dropdown</p> -->
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.html]: please remove commented lines

File: src/app/home/home-page/home-page.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<div class="header-section">
+    <h1>Home</h1>
+</div>
+<div class="page-layout">
+
```
Comment (miramax-88) [inline src/app/home/home-page/home-page.component.html]: please remove

File: src/app/home/filter-bar/filter-bar.component.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-filter-bar',
+  templateUrl: './filter-bar.component.html',
+  styleUrls: ['./filter-bar.component.scss']
+})
+export class FilterBarComponent {
+
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.ts:17]: please adjust

File: src/app/home/home-page/home-page.component.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
+
+.page-layout {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    padding: 0px;
+    gap: 10px;
+    background-color: rgb(238, 234, 234);
+
```
Comment (miramax-88) [inline src/app/home/home-page/home-page.component.scss]: please remove

File: src/app/home/home-page/home-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-home-page',
+  templateUrl: './home-page.component.html',
+  styleUrls: ['./home-page.component.scss']
+})
+export class HomePageComponent {
+
```
Comment (miramax-88) [inline src/app/home/home-page/home-page.component.ts:11]: please remove

File: src/app/home/table-data/table-data-datasource.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { map } from 'rxjs/operators';
+import { Observable, of as observableOf, merge } from 'rxjs';
+
+// TODO: Replace this with your own data model type
```
Comment (miramax-88) [inline src/app/home/table-data/table-data-datasource.ts]: here is example of how write TODO in google
 // TODO(b/112345): Replace this with your own data model type.

File: src/app/home/table-data/table-data-datasource.ts [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+export interface TableDataItem {
+  name: string;
+  id: number;
+}
+
+// TODO: replace this with real data from your application
```
Comment (miramax-88) [inline src/app/home/table-data/table-data-datasource.ts]: here is example of how write TODO in google
 // TODO(b/112345): Replace this with your own data model type.

File: src/app/home/table-data/table-data-datasource.ts [hunk 9]
```diff
@@ -0,0 +1,111 @@
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { map } from 'rxjs/operators';
+import { Observable, of as observableOf, merge } from 'rxjs';
+
+// TODO: Replace this with your own data model type
+export interface TableDataItem {
```
Comment (miramax-88) [inline src/app/home/table-data/table-data-datasource.ts:10]: for all exported entities please add comments. 

/**
 * Interface to work with blah blah blah
 */
export interface TableDataItem {

File: src/app/home/table-data/table-data-datasource.ts (hunk 8 above)
Comment (miramax-88) [reply src/app/home/table-data/table-data-datasource.ts]: and in other places if they exists

File: src/app/home/table-data/table-data.component.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import { AfterViewInit, Component, ViewChild } from '@angular/core';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { MatTable } from '@angular/material/table';
+import { TableDataDataSource, TableDataItem } from './table-data-datasource';
+
+@Component({
```
Comment (miramax-88) [inline src/app/home/table-data/table-data.component.ts:10]: please add comment about this class

File: src/app/home/filter-bar/filter-bar.component.ts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.ts:8]: please add comment to the class

File: src/app/home/home-page/home-page.component.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
```
Comment (miramax-88) [inline src/app/home/home-page/home-page.component.ts:6]: please add comment to the class

File: src/app/home/filter-bar/filter-bar.component.html (hunk 2 above)
Comment (rpendlimarri) [reply src/app/home/filter-bar/filter-bar.component.html]: done

File: src/app/home/home-page/home-page.component.html (hunk 3 above)
Comment (rpendlimarri) [reply src/app/home/home-page/home-page.component.html]: done

File: src/app/home/filter-bar/filter-bar.component.ts (hunk 4 above)
Comment (rpendlimarri) [reply src/app/home/filter-bar/filter-bar.component.ts:17]: done

File: src/app/home/table-data/table-data-datasource.ts (hunk 8 above)
Comment (rpendlimarri) [reply src/app/home/table-data/table-data-datasource.ts]: done

File: src/app/home/filter-bar/filter-bar.component.html [hunk 13]
```diff
@@ -0,0 +1,18 @@
+<div class="filter-bar-section">
+
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.html]: please remove empty lines

File: src/app/home/filter-bar/filter-bar.component.html [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+    <span>My App</span>
+    <span class="example-spacer"></span>
+    <button mat-icon-button class="example-icon favorite-icon" aria-label="Example icon-button with heart icon">
+      <mat-icon>favorite</mat-icon>
+    </button>
+    <button mat-icon-button class="example-icon" aria-label="Example icon-button with share icon">
+      <mat-icon>share</mat-icon>
+    </button>
+  </mat-toolbar>
+
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.html]: please remove empty line

File: src/app/home/filter-bar/filter-bar.component.ts (hunk 11 above)
Comment (miramax-88) [reply src/app/home/filter-bar/filter-bar.component.ts:8]: you should add comment above @Component({, 

and todo should look like this 
// TODO(b/12345): Class to work with Filter-bar and it's filters

File: src/app/home/filter-bar/filter-bar.component.ts [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+  selector: 'app-filter-bar',
+  templateUrl: './filter-bar.component.html',
+  styleUrls: ['./filter-bar.component.scss']
+})
+
+// (b/12345): Class to work with Filter-bar and it's filters
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.ts]: and todo should look like this 
// TODO(b/12345): Class to work with Filter-bar and it's filters

File: src/app/app.component.ts [hunk 16]
```diff
@@ -5,6 +5,8 @@ import { Component } from '@angular/core';
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
+/** (b/123234) : This component is responsible for handling the
```
Comment (miramax-88) [inline src/app/app.component.ts]: and todo should look like this 
// TODO(b/12345):

Comment (miramax-88) [reply src/app/app.component.ts]: /**
 * Component for ...
 */
@Component({

File: src/app/app.module.ts [hunk 17]
```diff
@@ -3,16 +3,22 @@ import { BrowserModule } from '@angular/platform-browser';
...
-    AppRoutingModule
+    AppRoutingModule,
+    HomeModule,
+    BrowserAnimationsModule
   ],
   providers: [],
   bootstrap: [AppComponent]
 })
+/** (b/232424) : This module holds the declarations, 
+imports and exports from the main app module */
```
Comment (miramax-88) [inline src/app/app.module.ts]: /**
 * Component for ...
 */
@Component({

File: src/app/app.module.ts [hunk 18]
```diff
@@ -3,16 +3,22 @@ import { BrowserModule } from '@angular/platform-browser';
...
     BrowserModule,
-    AppRoutingModule
+    AppRoutingModule,
+    HomeModule,
+    BrowserAnimationsModule
   ],
   providers: [],
   bootstrap: [AppComponent]
 })
+/** (b/232424) : This module holds the declarations,
```
Comment (miramax-88) [inline src/app/app.module.ts]: and todo should look like this
// TODO(b/12345):

File: src/app/home/home-page/home-page.component.ts (hunk 12 above)
Comment (miramax-88) [reply src/app/home/home-page/home-page.component.ts:6]: same here as above

File: src/app/home/home.module.ts [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+import { MatSlideToggleModule } from '@angular/material/slide-toggle';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { MatIconModule } from "@angular/material/icon";
+import { FilterBarComponent } from './filter-bar/filter-bar.component';
+import { HomePageComponent } from './home-page/home-page.component';
+import { TableDataComponent } from './table-data/table-data.component';
+import { MatTableModule } from '@angular/material/table';
+import { MatPaginatorModule } from '@angular/material/paginator';
+import { MatSortModule } from '@angular/material/sort';
+@NgModule({
```
Comment (miramax-88) [inline src/app/home/home.module.ts:24]: please add empty line

File: src/app/home/home.module.ts [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+    MatIconModule,
+    MatTableModule,
+    MatPaginatorModule,
+    MatSortModule
+  ],
+  exports: [
+    HomePageComponent
+  ]
+})
+// b(342332): This module is resonsible for holding the required modules for the Home page
```
Comment (miramax-88) [inline src/app/home/home.module.ts]: sorry in module it's not necessary to add comment even if it's export. 
so you should remove todo and comment

File: src/app/home/table-data/table-data-datasource.ts [hunk 21]
```diff
@@ -0,0 +1,111 @@
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { map } from 'rxjs/operators';
+import { Observable, of as observableOf, merge } from 'rxjs';
+
+// (b/345343): Interface to work with Table data item
```
Comment (miramax-88) [inline src/app/home/table-data/table-data-datasource.ts]: again misunderstanding how to write TODO and comments. We write TODO when necessary, but comments for each class and exported entities from class/utils

File: src/app/home/table-data/table-data-datasource.ts [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+export interface TableDataItem {
+  name: string;
+  id: number;
+}
+
+// (b/345343): This is the sample data to load into the Table
```
Comment (miramax-88) [inline src/app/home/table-data/table-data-datasource.ts]: If you want to add simple Comment, you can use //
Also you can follow rules from the doc I sent to you in slack

File: src/app/home/table-data/table-data-datasource.ts [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+  {id: 15, name: 'Phosphorus'},
+  {id: 16, name: 'Sulfur'},
+  {id: 17, name: 'Chlorine'},
+  {id: 18, name: 'Argon'},
+  {id: 19, name: 'Potassium'},
+  {id: 20, name: 'Calcium'},
+];
+
+/**
+ * (b/345343): Data source for the TableData view. This class should
```
Comment (miramax-88) [inline src/app/home/table-data/table-data-datasource.ts]: same here and bellow, please check all files.

File: src/app/home/filter-bar/filter-bar.component.ts [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+@Component({
+  selector: 'app-filter-bar',
+  templateUrl: './filter-bar.component.html',
+  styleUrls: ['./filter-bar.component.scss']
+})
+export class FilterBarComponent {
+  selectedOption1!: string;
+  selectedOption2!: string;
+  search = "";
+  constructor(public dialog: MatDialog) {}
```
Comment (miramax-88) [inline src/app/home/filter-bar/filter-bar.component.ts:18]: please add spaces around constructor.

**Review by miramax-88 — APPROVED** [review]
LGTM

Note please add dots (.) next time for any comment.
// This class is responsible to work with home page and it's child components >>**.**<<

**Review by miramax-88 — APPROVED** [review]

**Review by rpendlimarri — COMMENTED** [review]
Please review the changes. Thank you

**Review by miramax-88 — APPROVED** [review]

**Pull request merged**